core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -600,7 +604,20 @@ public void onFileInit(Path file) {
                         logger.warn("skipping compile of script file [{}] as all scripted operations are disabled for file scripts", file.toAbsolutePath());
                     }
                 } catch (Exception e) {
```

Comment:
Can you only do toxcontent for ScriptException, but still log the entire stack trace for other exceptions? eg, if we have an NPE because of a bug in painless, I think we should still see the full stack trace.